### ISSUE TITLE
Running deploy without ever running build unexpectedly deploys Hello Framework

### ISSUE THREAD

**Issue by mcglincy** [observablehq/framework#1214]
From @Fil: if you never call yarn build, but just yarn deploy immediately after you start editing files, it deploys "Hello Framework", with no warning that the build is stale.

**Comment by mbostock** [comment]
This is a high priority to fix.

**Comment by mbostock** [comment]
I added some logging, and I see this on a fresh deploy:

```
{
  buildFilePaths: [
    'index.html',
    '_file/observable.1af93621.png',
    '_observablehq/client.js',
    '_observablehq/runtime.js',
    '_observablehq/stdlib.js',
    '_observablehq/theme-air,near-midnight.css'
  ],
  youngestAge: 35954.673583984375,
  doBuild: false,
  BUILD_AGE_WARNING_MS: 300000
}
```

I think this means that `youngestAge <= BUILD_AGE_WARNING_MS`, and hence we don’t prompt you to build. But actually we’ve never built this project — this is the implicit build that happens during `observable create` whose primary purpose is to seed the npm cache #1145 — so we certainly shouldn’t consider this a “fresh” build. So, I think either we _always_ need to prompt regardless of date, or we check if any of the source files are more recent than the last build, and automatically build in that case, too.

**Comment by simon-mo** [comment]
Also ran into this when deploying my hello world, seems like working after I deleted `dist` and `docs/.observablehq/cache`. The framework is always great by the way, always something I wanted when building dashboard and really really enjoyed the data loader.

**Comment by mbostock** [comment]
Yep,  you can either build explicitly before deploying, or delete your build to workaround this problem. We have a fix in #1238 and will release it in the next few days.
